src/web/jcc/client/components/SetHubLocationPanel.tsx

```diff
@@ -0,0 +1,196 @@
+     */
+    function getInputHubLocation(): GeographicCoordinate {
+        const lat = Number(getElementById<HTMLInputElement>('set-hub-location-latitude').value)
+        const lon = Number(getElementById<HTMLInputElement>('set-hub-location-longitude').value)
+        if (lat == null || lon == null) {
+            return null
+        }
+
+        return {
+            lat: lat,
+            lon: lon
+        }
+    }
+
+
+    /**
+     * Calls the API to submit a location change for a certain hub.
+     *
+     */
+    function submitHubLocation() {
+        const hubLocation = getInputHubLocation()
+
+        if (hubLocation == null) {
+            console.warn('hub location is null')
+            return
+        }
+
+        const hubCommand: CommandForHub = {
+            hub_id: hub_id,
+            type: HubCommandType.SET_HUB_LOCATION,
+            hub_location: hubLocation
+        }
+
+        props.api.postCommandForHub(hubCommand)
+    }
+
+    
+    /**
+     * Initiate a PointerInteraction to select a new hub location with a click or tap.
+     */
+    const toggleSelectOnMapInteraction = () => {
+        if (selectOnMapInteractionRef.current !== null) {
+            destroySelectOnMapInteraction()
+            return
+        }
+
+        selectOnMapInteractionRef.current = new PointerInteraction({
+            handleEvent: (evt) => {
+                if (evt.type == "click") {
+                    const clickedLocation = getGeographicCoordinate(evt.coordinate, evt.map)
+                    getElementById<HTMLInputElement>('set-hub-location-latitude').value = clickedLocation.lat.toFixed(6)
+                    getElementById<HTMLInputElement>('set-hub-location-longitude').value = clickedLocation.lon.toFixed(6)
+                    submitHubLocation()
+                    destroySelectOnMapInteraction()
+                    // Return false to prevent other interactions from being affected by this click.
+                    return false
+                }
+                else {
+                    // Let this event fall through to the other interactions on the stack.
+                    return true
+                }
+            }
+        })
+        setSelectingOnMap(true)
+        props.map.addInteraction(selectOnMapInteractionRef.current)
+        props.map.getTargetElement().style.cursor = 'crosshair'
+    }
+    
+
+    /**
+     * Destroy the hub location selection interaction.
+     */
+    const destroySelectOnMapInteraction = () => {
+        if (selectOnMapInteractionRef.current !== null) {
+            props.map.removeInteraction(selectOnMapInteractionRef.current)
+            selectOnMapInteractionRef.current = null
+            props.map.getTargetElement().style.cursor = 'default'
+            setSelectingOnMap(false)
+        }
+    }
+
+
+    // Destroy the map select interaction on unmount, if present.
+    useEffect(() => {
+        return destroySelectOnMapInteraction
+    }, [])
+
+
+    return (
```

Comment:
Hey Ed, can we move the inline css into its own css file. A SetHubLocation.css or .less file in style/components. A less file would probably be better so we can use the nesting to avoid naming collisions with other files.

Reply:
Yeah, we can certainly do that.  I did it in an ad-hoc way here, only because I had to finagle it to get things to center properly.  It's kind of a code smell for our css, that we need to massage the styles so much to get them to look good.  I was trying to localize those tweaks, so they didn't affect any other elements... but yes they're probably better off in a less file for that panel.